### PR TITLE
Added instagram filter cookbook recipe.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Added instagram filter cookbook recipe.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@RedBrogdon can you review this code for appropriate brevity and visual effects for recipe purposes?

File: cookbook/lib/examples/instagram_filter.dart [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    );
+  }
+
+  Widget _buildPhotoWithFilter() {
+    return ValueListenableBuilder(
+      valueListenable: _filterColor,
+      builder: (context, value, child) {
+        final color = value as Color;
+        return Image.network(
+          'https://flutter.dev/docs/cookbook/img-files/effects/instagram-buttons/millenial-dude.jpg',
```
Comment (RedBrogdon) [inline cookbook/lib/examples/instagram_filter.dart:70]: nit: This line length will give the CodeMirror editor in DartPad some problems. Two adjacent string literals would get around it here and in the other image further down.

**Review by RedBrogdon — APPROVED** [review]
LGTM other than the one DartPad-related nit.

Great work slimming this code down, BTW. Having seen the other version, I can tell there's a difference, but I'd never feel anything was missing if I came to this one on tis own.

File: cookbook/lib/examples/instagram_filter.dart (hunk 1 above)
Comment (domesticmouse) [reply cookbook/lib/examples/instagram_filter.dart:70]: May be better to pull `https://flutter.dev/docs/cookbook` out as a constant, as I'm guessing you will be re-using it. 

I'm going to land this as is and let the nits be cleaned up in a follow up PR

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**